services/app-web/src/pages/SubmissionSideNav/SubmissionSideNav.tsx

```diff
@@ -54,8 +81,80 @@ export const SubmissionSideNav = () => {
         return getRouteName(pathname) === route ? 'usa-current' : ''
     }
 
-    const { result: fetchResult } = useFetchHealthPlanPackageWrapper(id)
-    if (fetchResult.status === 'LOADING') {
+    const parseQuestions = async (
+        questions: IndexQuestionsPayload
+    ): Promise<QADivisionQuestions> => {
+        const extractQuestions = async (
+            edges: QuestionEdge[]
+        ): Promise<QuestionData[]> => {
+            const questions = await Promise.all(
+                edges.map(async ({ node }) => ({
+                    id: node.id,
+                    pkgID: node.pkgID,
+                    createdAt: node.createdAt,
+                    addedBy: node.addedBy as CmsUser,
+                    documents: (await getDocumentsUrl(
+                        node.documents,
+                        'QUESTION_ANSWER_DOCS'
+                    )) as QuestionDocumentWithLink[],
+                }))
+            ).catch((err) => {
+                console.info(err)
+                return []
+            })
+
+            return questions
+        }
+
+        const divisionQuestions = {
+            dmco: {
+                totalCount: questions.DMCOQuestions.totalCount ?? 0,
+                questions: await extractQuestions(
+                    questions.DMCOQuestions.edges
+                ),
+            },
+            dmcp: {
+                totalCount: questions.DMCPQuestions.totalCount ?? 0,
+                questions: await extractQuestions(
+                    questions.DMCPQuestions.edges
+                ),
+            },
+            oact: {
+                totalCount: questions.OACTQuestions.totalCount ?? 0,
+                questions: await extractQuestions(
+                    questions.OACTQuestions.edges
+                ),
+            },
+        }
+
+        return divisionQuestions
+    }
+
+    const { result: fetchResult } =
+        useFetchHealthPlanPackageWithQuestionsWrapper(id, async (data) => {
```

Comment:
I’m surprised you’re using the callback here instead of parsing questions out of the result data below like we get the package results etc.